drivers: sam0_rtc_timer: DT_INST changes have broken this driver

**Describe the bug**
Commit 28870e7a32dba06203c9457dfcd03e0aa04fb130 appears to have broken the sam0_rtc_timer driver.  The new macro appears to be generating a name that doesn't exist in the devicetree generated file.  I think this problem is related to #23107 and #22806

Here is the error message:
```
/Users/stevenslupsky/Documents/source/zephyrproject/zephyr/drivers/timer/sam0_rtc_timer.c:75:22: error: 'GCLK_CLKCTRL_GEN_GCLKDT_N_S_soc_S_rtc_40001400_P_clocks_IDX_0_VAL_clock_generator' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_rtc_40001400_P_clock_generator'?
```
From the generated file:
```
/* Generic property macros: */
#define DT_N_S_soc_S_rtc_40001400_P_reg {1073746944 /* 0x40001400 */, 28 /* 0x1c */}
#define DT_N_S_soc_S_rtc_40001400_P_reg_IDX_0 1073746944
#define DT_N_S_soc_S_rtc_40001400_P_reg_IDX_1 28
#define DT_N_S_soc_S_rtc_40001400_P_reg_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_clock_generator 4
#define DT_N_S_soc_S_rtc_40001400_P_clock_generator_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_label "RTC"
#define DT_N_S_soc_S_rtc_40001400_P_label_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_interrupts {3 /* 0x3 */, 0 /* 0x0 */}
#define DT_N_S_soc_S_rtc_40001400_P_interrupts_IDX_0 3
#define DT_N_S_soc_S_rtc_40001400_P_interrupts_IDX_1 0
#define DT_N_S_soc_S_rtc_40001400_P_interrupts_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_prescaler 0
#define DT_N_S_soc_S_rtc_40001400_P_prescaler_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_status "okay"
#define DT_N_S_soc_S_rtc_40001400_P_status_ENUM_IDX 1
#define DT_N_S_soc_S_rtc_40001400_P_status_EXISTS 1
#define DT_N_S_soc_S_rtc_40001400_P_compatible {"atmel,sam0-rtc"}
#define DT_N_S_soc_S_rtc_40001400_P_compatible_IDX_0 "atmel,sam0-rtc"
#define DT_N_S_soc_S_rtc_40001400_P_compatible_LEN 1
#define DT_N_S_soc_S_rtc_40001400_P_compatible_EXISTS 1
```
It appears as if the wrong macro is being used here but I could not find a macro in devicetree/clocks.h that would generate the correct result.  So, I am not sure how to fix this.

--- a/drivers/timer/sam0_rtc_timer.c
+++ b/drivers/timer/sam0_rtc_timer.c
@@ -186,7 +186,7 @@ int z_clock_driver_init(struct device *device)
 	/* Set up bus clock and GCLK generator. */
 	PM->APBAMASK.reg |= PM_APBAMASK_RTC;
 	GCLK->CLKCTRL.reg = GCLK_CLKCTRL_ID(RTC_GCLK_ID) | GCLK_CLKCTRL_CLKEN
-			    | GCLK_GEN(DT_INST_CLOCKS_CELL(0, generator));
+			    | GCLK_GEN(DT_INST_PROP(0, clock_generator));
 
 	/* Synchronize GCLK. */
 	while (GCLK->STATUS.bit.SYNCBUSY) {

